web/themes/custom/usagov/scripts/modal.js

```diff
@@ -0,0 +1,40 @@
+
+function processModalUrlParameter() {
+    "use strict";
+
+    const urlParams = new URLSearchParams(window.location.search);
+    const modalName = urlParams.get('modal');
+    if (modalName === null) return;
+
+    toggleModal(modalName);
+
+    urlParams.delete("modal");
+    let newURL = window.location.pathname;
+    if (urlParams.size) {
+        newURL = newURL + '?' + urlParams.toString();
+    }
+    history.replaceState(history.state, null, newURL);
+}
+
+/* open or close the modal programatically */
+function toggleModal(modalName) {
+    "use strict";
+
+    let map = document.querySelector('[data-modal-name="'+ modalName +'"]');
+    if (map === null) return;
+    let modalID = map.getAttribute('data-modal-id');
+    if (modalID === null) return;
+    document.querySelector('[href="#paragraph--id--'+ modalID +'"]').click();
+    document.querySelector('#paragraph--id--'+ modalID).querySelector('button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])').focus();
+}
+
+document.addEventListener('DOMContentLoaded', function() {
+    "use strict";
+
+    document.querySelectorAll('.usa-modal').forEach((modal) => {
```

Comment:
These are great questions.

### Why set modal.hidden to false:

I was seeing a flash of modal content while the page was loading (similar to the recent gov banner issue), so I made [this commit](https://github.com/usagov/usagov-2021/pull/1896/commits/51b54b2398dd65e6ede9fb613fa6139df665cfc0) to ensure that they start hidden while the page loads. Eventually, the USWDS script hides the modals a different way so we can safely remove our hidden state. Since it is only related to preventing a flash of content, I chose to keep it outside the url parameter function.

### The toggle function:

I agree about the toggle concern, so I've rewritten the script with open and close functions. I also refined everything and added comments for clarity. It still relies on a toggle function because as far as I can tell the USWDS does not provide open and close javascript functions so our options are to trigger their code by clicking their toggle button or try to mimic their code and potentially miss something important.

Reply:
Thanks, I understand now. I like your toggle solution.